frontend/components/forms/DynamicForm.tsx

```diff
@@ -0,0 +1,226 @@
+import TextField from "@mui/material/TextField";
+import { useRouter } from "next/router";
+import { ParsedUrlQueryInput } from "querystring";
+import { FC, MutableRefObject, useContext, useEffect, useRef, useState } from "react";
+import PageTransitionContext from "../PageTransitionContext";
+
+const DynamicForm: FC = () => {
+  const [type, setType] = useState("");
+
+  const handleFormType = (event: { target: { value: string } }) => {
+    setType(event.target.value);
+  };
+
+  return (
+    <Card elevation={5}>
+      <FormControl sx={{ minWidth: "20rem", m: "1rem" }}>
+        <InputLabel id="select-form-type">Type of Content</InputLabel>
+        <Select
+          labelId="select-form-type"
+          id="select-form"
+          value={type}
+          label="Type of Content"
+          onChange={handleFormType}
+        >
+          <MenuItem value={"none"}>
+            <em>None</em>
+          </MenuItem>
+          <MenuItem value={"quotes"}>Quote</MenuItem>
+          <MenuItem value={"entities"}>Entity</MenuItem>
+          <MenuItem value={"images"}>Image</MenuItem>
+          <MenuItem value={"occurrences"}>Occurrence</MenuItem>
+          <MenuItem value={"topics"}>Topic</MenuItem>
+        </Select>
+      </FormControl>
+      {type && type != "none" && (
+        <>
+          <Divider sx={{ borderBottomWidth: 2 }} />
+          <Box
+            sx={{
+              flex: "1 1",
+              display: "flex",
+              alignItems: "center",
+              justifyContent: "center",
+              margin: "1.5rem 1rem 1.5rem 1rem",
+            }}
+          >
+            <Box>
+              <DynamicFormFields type={type} />
+            </Box>
+          </Box>
+        </>
+      )}
+    </Card>
+  );
+};
+
+interface DynamicFormProps {
+  type: string;
+}
+
+const fields = ["topics"] as const;
+type Field = typeof fields[number];
+type FieldsRef = MutableRefObject<Record<Field, any>>;
+type FieldCallbacks = Record<Field, (value: ParsedUrlQueryInput[string]) => void>;
+
+//Dynamic fields form
+//Can be called using the DynamicForm component or by using the DynamicFormFields prop and specifying the 'type' prop
+const DynamicFormFields: FC<DynamicFormProps> = ({ type }: DynamicFormProps) => {
+  const [formData, setFormData] = useState([]);
+  const [choiceValue, setChoiceValue] = useState<string[]>([]);
+
+  const getDynamicFields = async (type: string) => {
+    return await axiosWithoutAuth.get(`/api/${type}/fields/`).then((response) => {
+      return response.data;
+    });
+  };
+
+  const router = useRouter();
+  const isLoading = useContext(PageTransitionContext);
+
+  const handleChoice = (event: { target: { value: string[] | string } }) => {
+    const value = event.target.value as string[] | string;
+    setChoiceValue(typeof value === "string" ? value.split(",") : value);
+  };
+
+  const fieldsRef = useRef(
+    Object.fromEntries(fields.map((name) => [name, router.query[name]]))
+  ) as FieldsRef;
```

Comment:
Do we want/need to initialize field values from URL parameters? At least right now they are all undefined, so we may as well initialize the useRef with {}.